tools/blocksconvert/builder/tsdb.go

```diff
@@ -166,31 +172,31 @@ func (d *tsdbBuilder) buildSingleSeries(metric labels.Labels, cs []chunk.Chunk)
 	minTime := chs[0].MinTime
 	maxTime := chs[len(chs)-1].MaxTime
 
-	d.writtenSamples.Add(float64(seriesSamples))
-	d.addSeries(metric, chs, seriesSamples, minTime, maxTime)
-	return nil
-}
-
-func (d *tsdbBuilder) addSeries(m labels.Labels, cs []chunks.Meta, samples uint64, minTime, maxTime int64) {
-	sl := d.symbolsMap.toSymbolLabels(m)
-
-	d.seriesMu.Lock()
-	defer d.seriesMu.Unlock()
+	inmem, err := d.series.addSeries(metric, chs, seriesSamples, minTime, maxTime)
```

Comment:
Design wise, looks a bit weird to me that `addSeries()` returns the count of total series. At the cost of taking the read lock another time, I would probably add `d.series.count()` (or something similar) to update the `seriesInMemory` metric.